controllers/consumerCouponController.js

```diff
@@ -0,0 +1,109 @@
+const ConsumerCoupon = require("../models/ConsumerCoupon");
+
+const getAllConsumerCoupons = async (req, res) => {
+  try {
+    const consumerId = req.consumerId;
+
+    if (!consumerId) {
+      return res.status(404).json({ error: "Consumer not found" });
+    }
+
+    // Find all consumer coupons for the consumer and populate consumer and coupon fields
+    const consumerCoupons = await ConsumerCoupon.find({ consumer: consumerId })
+      .populate({
+        path: "consumer",
+        populate: { path: "user", select: "-password" },
+      })
+      // .populate({
+      //   path: "coupon",
+      //   populate: {
+      //     path: "merchant",
+      //     populate: {
+      //       path: "user",
+      //       select: "-password",
+      //     },
+      //   },
+      // })
+      .lean();
+
+    // Return the cnsumerCoupons
+    res.json({ consumerCoupons });
+  } catch (error) {
+    // Handle errors
+    console.error(error);
+    res.status(500).json({ error: "Internal server error" });
+  }
+};
+
+const addConsumerCoupon = async (req, res) => {
+  try {
+    const consumerId = req.consumerId;
+
+    if (!consumerId) {
+      return res.status(404).json({ error: "Consumer not found" });
+    }
+
+    const { couponId } = req.body;
+
+    if (!couponId) {
+      return res.status(400).json({ message: "couponId is required" });
+    }
+
+    const newConsumerCoupon = await ConsumerCoupon.create({
+      consumer: consumerId,
+      coupon: couponId,
+      qrCode: "qrCode",
+      qrIdentification: "qrIdentification",
+      status: "active",
+    });
```

Comment:
Use the qr code base64 generator, and then create an identifier so that the merchant can use the identification when qrCode fails.